spring-boot/admin/src/main/java/eu/coatrack/admin/service/OAuthUserDetailsService.java

```diff
@@ -56,19 +73,29 @@ private OAuth2User getLoggedInUser() {
     }
 
     public String getLoginNameFromLoggedInUser() {
-        return getLoggedInUser().getAttribute("login");
+        return getLoggedInUser().getAttribute(attributeNameForLogin);
     }
 
     public String getNameFromLoggedInUser() {
-        return getLoggedInUser().getAttribute("name");
+        return getLoggedInUser().getAttribute(attributeNameForName);
     }
 
     public String getCompanyFromLoggedInUser() {
-        return getLoggedInUser().getAttribute("company");
+        if ((attributeNameForCompany).equals(KEYROCK_COMPANY_ATTRIBUTE)) {
```

Comment:
your basic question here is: "are we currently using Keyrock". This should not be derived from the name of an attribute, because in the future we could use other Oauth providers that use the same attribute for companies. 

Ideas for another way:
- either get the currently used spring profile to decide if we are using keyrock
- or maybe you can somehow get the oauth provider name from the config?
- or add a new config attribute e.g. `ygg.admin.idm.provider` that contains `github` or `keyrock`

Reply:
This is already being used in the PR contained the "Add API key to users", and the only usage in this class "OauthUserDetailsService" is for the company, so if that code is dropped, then this thread is resolved

Reply:
Done